### ISSUE TITLE
Compiler (gcc 10.1.0) warning in: src/gldrv/gl_texture.cpp

### ISSUE THREAD

**Issue by evertvorster** [vegastrike/Vega-Strike-Engine-Source#256]
[ 41%] Building CXX object CMakeFiles/vegastrike.dir/src/gldrv/gl_texture.cpp.o
/home/evert/vsmod/git/Vega-Strike-Engine-Source/engine/src/gldrv/gl_texture.cpp: In function 'unsigned char GFXCreateTexture(int, int, TEXTUREFORMAT, int*, char*, int, FILTER, TEXTURE_TARGET, ADDRESSMODE)':
/home/evert/vsmod/git/Vega-Strike-Engine-Source/engine/src/gldrv/gl_texture.cpp:348:12: warning: enumeration value 'TEXTURE3D' not handled in switch [-Wswitch]
  348 |     switch (texture_target)
      |            ^
[ 41%] Building CXX object CMakeFiles/vegastrike.dir/src/gldrv/gl_vertex_list.cpp.o

**Comment by evertvorster** [comment]
OK, the code block in question looks like this:
    switch (texture_target)
    {
    case TEXTURE1D:
        textures[*handle].targets = GL_TEXTURE_1D;
        break;
    case TEXTURE2D:
        textures[*handle].targets = GL_TEXTURE_2D;
        break;
#ifdef GL_EXT_texture3D
    case TEXTURE3D:
        textures[*handle].targets = GL_TEXTURE_3D;
        break;
#endif
    case CUBEMAP:
        textures[*handle].targets = GL_TEXTURE_CUBE_MAP_EXT;
        break;
    case TEXTURERECT:
        textures[*handle].targets = GL_TEXTURE_RECTANGLE_ARB;
        break;
    }

In this function the preprocessor chops out the bit that gcc is complaining about. Right now it feels like this was done to prevent a crash somewhere, but without a default, what is the point of leaving it undefined?

**Comment by evertvorster** [comment]
OK, after hunting around a bit, "GL_EXT_texture3D" entered the OpenGL spec around 1996. 
According to this gem I found: https://www.gamedev.net/forums/topic/332346-gf4-and-gl_ext_texture3d/
In 2005 a dated video card already had hardware support for this feature, and the drivers just needed to be updated. 

TL;DR - we can safely assume that every user now supports this, and drop the #ifdef.

**Comment by evertvorster** [comment]
Fixed by #261
